### PR TITLE
Adding notebook with Water Facility Network functionality

### PR DESCRIPTION
**Pull request by nunezco2**

### CODE REVIEW

**Comment by navarroc** [comment]
You need to add references to manual_jb/content/notebooks_other.md.

File: manual_jb/content/notebooks_other.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ Note that some Notebooks might not work with the latest version of pyIncore.
 [tornadoepn_dmg.ipynb](https://github.com/IN-CORE/incore-docs/blob/master/notebooks/tornadoepn_dmg.ipynb) <br />
 [transportation_recovery.ipynb](https://github.com/IN-CORE/incore-docs/blob/master/notebooks/transportation_recovery.ipynb) <br />
 [water_facility_dmg.ipynb](https://github.com/IN-CORE/incore-docs/blob/master/notebooks/water_facility_dmg.ipynb) <br />
+[wfn_functionality.ipynb](https://github.com/IN-CORE/incore-docs/blob/master/notebooks/wfn_functionality.ipynb) <br />
```
Comment (navarroc) [inline manual_jb/content/notebooks_other.md:37]: this should be main not master. The link will work once we release.

Comment (navarroc) [reply manual_jb/content/notebooks_other.md:37]: I have an issue to fix the others that should also be main and not master. I will do that right before we release.

**Comment by nunezco2** [comment]
> Will you be adding the analysis documentation in a separate PR or this one? We need to create a water_functionality.md file similar to what we did with this one for pipeline functionality: #132

Done

**Comment by nunezco2** [comment]
> You need to add references to manual_jb/content/notebooks_other.md.

Done

**Comment by navarroc** [comment]
> > Will you be adding the analysis documentation in a separate PR or this one? We need to create a water_functionality.md file similar to what we did with this one for pipeline functionality: #132
> 
> Done

The analysis documentation also needs to be linked from the side bar using the _toc.yml and from the analyses.md

File: manual_jb/content/_toc.yml [hunk 2]
```diff
@@ -44,6 +44,7 @@ chapters:
       - file: analyses/tornadoepn_dmg
       - file: analyses/transportation_recovery
       - file: analyses/waterfacility_dmg
+      - file: analyses/wfn_functionality.dmg
```
Comment (navarroc) [inline manual_jb/content/_toc.yml:47]: Remove .dmg

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ## [Unreleased]
 
+ ### Added
+ - Water facility nework functionality [#139](https://github.com/IN-CORE/incore-docs/issues/139).
```
Comment (navarroc) [inline CHANGELOG.md:10]: typo in network

File: manual_jb/content/notebooks_other.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (navarroc) [reply manual_jb/content/notebooks_other.md:37]: This should be: https://github.com/IN-CORE/incore-docs/blob/main/notebooks/wfn_functionality.ipynb

File: manual_jb/content/analyses/wfn_functionality.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
+# Water facility restoration
```
Comment (navarroc) [inline manual_jb/content/analyses/wfn_functionality.md:1]: This should be Water Facility Network Functionality, right?

File: manual_jb/content/analyses/wfn_functionality.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    wfn_func.set_input_dataset("pp_sample_failure_state", pp_dmg_fs)
+    wfn_func.set_parameter("result_name", "mmsa_wfn_functionality")
+    wfn_func.set_parameter("tank_node_list", [1, 7, 10, 13, 14, 15])
+    wfn_func.set_parameter("pumpstation_node_list", [2, 3, 4, 5, 6, 8, 9, 11, 12])
+
+    # Run Analysis
+    wfn_func.run_analysis()
+```
+
+full analysis: [wfn_functionality.ipynb](https://github.com/IN-CORE/incore-docs/blob/master/notebooks/wfn_functionality.ipynb)
```
Comment (navarroc) [inline manual_jb/content/analyses/wfn_functionality.md:78]: Can you change "master" to "main"?

File: notebooks/wfn_functionality.ipynb [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+    "wfn_func.load_remote_input_dataset(\"wfn_network\", wfn_dataset_id)\n",
+    "wfn_func.set_input_dataset(\"wf_sample_failure_state\", wf_dmg_fs)\n",
+    "wfn_func.set_input_dataset(\"pp_sample_failure_state\", pp_dmg_fs)\n",
+    "wfn_func.set_parameter(\"result_name\", \"mmsa_wfn_functionality\")\n",
+    "wfn_func.set_parameter(\"tank_node_list\", [1, 7, 10, 13, 14, 15])\n",
+    "wfn_func.set_parameter(\"pumpstation_node_list\", [2, 3, 4, 5, 6, 8, 9, 11, 12])\n",
+    "\n",
+    "wfn_func.run_analysis()\n",
+    "\n",
+    "df = wfn_func.get_output_dataset('result').get_dataframe_from_csv()\n",
```
Comment (navarroc) [inline notebooks/wfn_functionality.ipynb:301]: "result" isn't the right key for this analysis

**Review by navarroc — APPROVED** [review]

File: manual_jb/content/analyses/wfn_functionality.md [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+
+**Related publications**
+
+* Sharma, N., Tabandeh, A., & Gardoni, P. (2019). Regional resilience analysis: A multi-scale approach to model the 
+  recovery of interdependent infrastructure. In P. Gardoni (Ed.), *Handbook of sustainable and resilient 
+  infrastructure* (pp. 521–544). New York, NY: Routledge.
+* Sharma, N., Tabandeh, A., & Gardoni, P. (2020). Regional resilience analysis: A multi-scale approach to optimize 
+  the resilience of interdependent infrastructure. *Computer‐Aided Civil and Infrastructure Engineering*, **35(12)**, 
+  1315-1330.
+* harma, N., & Gardoni, P. (2022). Mathematical modeling of interdependent infrastructure: An object-oriented
```
Comment (longshuicy) [inline manual_jb/content/analyses/wfn_functionality.md:27]: Typo? Missing a S? Sharma?

Comment (navarroc) [inline manual_jb/content/analyses/wfn_functionality.md:27]: @nunezco2 missing S in the citation.

File: manual_jb/content/analyses/wfn_functionality.md [hunk 8]
```diff
@@ -24,7 +24,7 @@ with functionality failure states.
 * Sharma, N., Tabandeh, A., & Gardoni, P. (2020). Regional resilience analysis: A multi-scale approach to optimize 
   the resilience of interdependent infrastructure. *Computer‐Aided Civil and Infrastructure Engineering*, **35(12)**, 
   1315-1330.
-* harma, N., & Gardoni, P. (2022). Mathematical modeling of interdependent infrastructure: An object-oriented 
+* Sßharma, N., & Gardoni, P. (2022). Mathematical modeling of interdependent infrastructure: An object-oriented
```
Comment (navarroc) [inline manual_jb/content/analyses/wfn_functionality.md:27]: @nunezco2 There is an odd character now after the S

File: manual_jb/content/analyses/wfn_functionality.md (hunk 7 above)
Comment (navarroc) [reply manual_jb/content/analyses/wfn_functionality.md]: @longshuicy can you take another look, I think this issue is addressed.

**Review by longshuicy — APPROVED** [review]

**Review by ywkim312 — APPROVED** [review]
ran okay

**Pull request merged**